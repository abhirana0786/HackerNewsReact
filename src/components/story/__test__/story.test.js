import React from 'react';
import ReactDOM from 'react-dom';
import StoryComponent from './../story';
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer'

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<StoryComponent></StoryComponent>, div)
});

it("renders story component correctly", () => {
    render(<StoryComponent></StoryComponent>)
});

it("matches snapshot", () => {
    const tree = renderer.create(<StoryComponent></StoryComponent>).toJSON();
    expect(tree).toMatchSnapshot();
});