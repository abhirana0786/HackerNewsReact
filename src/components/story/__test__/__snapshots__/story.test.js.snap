// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <table
    border="0"
    cellPadding="0"
    cellSpacing="0"
    id="storyTable"
    width="100%"
  >
    <thead>
      <tr>
        <th>
          Comments
        </th>
        <th>
          Vote Count
        </th>
        <th>
          UpVote
        </th>
        <th
          width="80%"
        >
          News Details
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="pagination"
  >
    <a
      className="pointer"
      onClick={[Function]}
    >
      Previous
    </a>
     | 
    <a
      className="pointer"
      onClick={[Function]}
    >
      Next
    </a>
  </div>
  <div
    className="chart-view"
  />
</div>
`;
